.github/workflows/generate-website.yml

```diff
@@ -0,0 +1,43 @@
+name: Generate Website
+
+on:
+  pull_request:
+    types: [opened, synchronize, reopened, ready_for_review]
+  push:
+    branches:
+      - master
+
+jobs:
+
+  generate-website:
+    name: Generate Website
+    runs-on: ubuntu-latest
+    steps:
+
+    - name: Checkout
+      uses: actions/checkout@v3
+
+    - name: Build Jekyll
+      uses: actions/jekyll-build-pages@v1
+
+    - name: Install exhale and doxygen
+      working-directory: ${{github.workspace}}
+      run: |
+        pip install sphinx sphinx-book-theme exhale
+        sudo apt-get install doxygen
```

Comment:
do you need sudo? I think you are already root here.

Reply:
nop, I'm not root, but the sudo is passwordless.